tests/rados/test_bluestore_features.py

```diff
@@ -0,0 +1,94 @@
+"""
+This file contains the  methods to verify the Bluestore features.
+"""
+
+
+import datetime
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from tests.rados.monitor_configurations import MonConfigMethods
+from utility.log import Log
+from utility.utils import convert_bytes, get_storage_stats, method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Performs various pool related validation tests
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_object = RadosOrchestrator(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_object)
+    client_node = ceph_cluster.get_nodes(role="client")[0]
+
+    target_configs = config["cache_trim_max_skip_pinned"]["configurations"]
+    max_skip_pinned_value = int(
+        mon_obj.get_config(section="osd", param="bluestore_cache_trim_max_skip_pinned")
+    )
+
+    # Check the default value of the  bluestore_cache_trim_max_skip_pinned value
+    if max_skip_pinned_value != 1000:
+        log.error(
+            "The default value of  bluestore_cache_trim_max_skip_pinned not equal to 1000"
+        )
+        return 1
+
+    # Creating pools and starting the test
+    for entry in target_configs.values():
+        log.debug(
+            f"Creating {entry['pool_type']} pool on the cluster with name {entry['pool_name']}"
+        )
+    method_should_succeed(
+        rados_object.create_pool,
+        **entry,
+    )
+
+    if not rados_object.bench_write(**entry):
+        log.error("Failed to write objects into the EC Pool")
+        return 1
+    rados_object.bench_read(**entry)
+    log.info("Finished writing data into the pool")
+    rados_object.bench_write(**entry)
+
+    # performing scrub and deep-scrub
+    rados_object.run_scrub()
+    rados_object.run_deep_scrub()
+    time.sleep(10)
+
+    # Checking for the memory growth
+    cluster_stats = get_storage_stats(client_node)
```

Comment:
Hey Bharath, I think there is a confusion here. We are supposed to check the memory consumption growth of the OSD daemons, not the ceph cluster storage stats.

Can you please check again? Similar activity is needed for task : RHCEPHQE-7917 . Ref : https://access.redhat.com/articles/351143

Reply:
Code modified and explained above.